File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-1.cpp
class Solution {
public:
int solve(int low, int high, int target, vector<int>&arr){
    while(low<=high){
        int mid=low+(high-low)/2;
        cout<<mid<<endl;
        if(arr[mid]==target){
            return mid;
        }
        else if(arr[low]<=arr[mid]){
            if(arr[low]<=target && target<=arr[mid]){
                high=mid-1;
            }
            else{
                low=mid+1;
            }
        }
        else{
            if(arr[mid]<=target && target<=arr[high]){
                low=mid+1;
            }
            else{
                high=mid-1;
            }
        }
    }

    return -1;
}
    int search(vector<int>& nums, int target) {
        int n=nums.size();
       return solve(0,n-1,target, nums); 
    }
};

File: Data Structures & Algorithms/combination-target-sum/submission-1.cs
public class Solution {

    void solve(int i, int n, int target, List<int>path, int[] nums, List<List<int>> ans){
        if(target==0){
            ans.Add(new List<int>(path));
            return;
        }

        if(i==n) return;

        if(nums[i]<=target){
            path.Add(nums[i]);
            solve(i, n, target-nums[i], path, nums, ans);
            path.Remove(nums[i]);
        }

        solve(i+1, n, target, path, nums, ans);
        return;
    }
    public List<List<int>> CombinationSum(int[] nums, int target) {
        List<List<int>>ans=new List<List<int>>();
        List<int>path = new List<int>();
        int n=nums.Length;

        solve(0,n,target, path,nums, ans);
        return ans;

    }
}
